### PR TITLE
Adds flag and policy to require all buckets requests to be over SSL

### PR DESCRIPTION
**Pull request by bradj**
## what

Adds flag that requires all S3 requests to be over SSL

## why

Compliance and security

## references

n/a

### CODE REVIEW

**Review by Benbentwo — APPROVED** [review]

File: main.tf [hunk 1]
```diff
@@ -129,6 +131,28 @@ data "aws_iam_policy_document" "default" {
     }
   }
 
+  dynamic "statement" {
+    for_each = var.allow_ssl_requests_only ? [1] : []
+
+    content {
+      sid       = "ForceSSLOnlyAccess"
```
Comment (osterman) [inline main.tf:138]: I feel like the language of the Sid should match the language of the feature flag

Comment (bradj) [reply main.tf]: @osterman updated

**Review by Benbentwo — APPROVED** [review]

File: main.tf [hunk 2]
```diff
@@ -129,6 +131,28 @@ data "aws_iam_policy_document" "default" {
...
   }
 
+  dynamic "statement" {
+    for_each = var.allow_ssl_requests_only ? [1] : []
+
+    content {
+      sid       = "AllowSSLRequestsOnly"
+      effect    = "Deny"
+      actions   = ["s3:*"]
+      resources = [local.bucket_arn, "${local.bucket_arn}/*"]
```
Comment (aknysh) [inline main.tf:141]: @bradj  we should have updated the bucket ARNs in the other places as well


```
resources = ["${aws_s3_bucket.default[0].arn}/*"]
```

**Pull request merged**